pkg/agent/agent.go

```diff
@@ -1160,3 +1160,35 @@ func (i *Initializer) initVMLocalConfig(nodeName string) error {
 	}
 	return nil
 }
+
+// prepareOVSBridge operates OVS bridge.
+func (i *Initializer) prepareOVSBridge() error {
+	if i.nodeType == config.K8sNode {
+		return i.prepareOVSBridgeForK8sNode()
+	}
+	return i.prepareOVSBridgeForVM()
+}
+
+func (i *Initializer) prepareOVSBridgeForVM() error {
+	return i.setOVSDatapath()
+}
+
+// setOVSDatapath generates a static datapath ID for OVS bridge so that the OFSwitch identifier is not
+// changed after the physical interface is attached on the switch.
+func (i *Initializer) setOVSDatapath() error {
+	otherConfig, err := i.ovsBridgeClient.GetOVSOtherConfig()
+	if err != nil {
+		klog.ErrorS(err, "Failed to read OVS bridge other_config")
+		return err
+	}
+	if _, exists := otherConfig[ovsconfig.OVSOtherConfigDatapathIDKey]; exists {
+		return nil
+	}
```

Comment:
should it use `i.ovsBridgeClient.GetDatapathID()`?

Reply:
No, it doesn't work. When setting OVS datapath_id, we actually fill the "other_config" field to tell OVS the expected datapath ID. If we don't do it, OVS would generate a random id on field "datapath_id", where `i.ovsBridgeClient.GetDatapathID()` is read from. That is to say, the return value of function `i.ovsBridgeClient.GetDatapathID()` is always not empty. But if other_config is not set, the value in datapath_id would change when we add the uplink to OVS, and this is possible to introduce a disconnection from Agent to OVS bridge. 

As for this code, it is for restart case, and we should check that a value is already set by Antrea Agent but not allocated by OVS. So we should read other_config field here.

Reply:
got it, thanks for the explanation. maybe worth to add a comment to explain it to avoid people like me trying to "optimize" it by resuing the `GetDatapathID` method.